app/atomic_svc.py

```diff
@@ -0,0 +1,253 @@
+        Populate internal dictionary used to match techniques to corresponding tactics.
+        By default, use the file 'enterprise-attack.json' located in the Atomic Red Team repository.
+        You can use a specific version of this file by passing its path with the `entreprise_attack_path`
+        parameter.
+        '''
+        self.technique_to_tactics = defaultdict(list)
+        if not entreprise_attack_path:
+            entreprise_attack_path = os.path.join(self.repo_dir, 'atomic_red_team', 'enterprise-attack.json')
+
+        with open(entreprise_attack_path, 'r') as f:
+            mitre_json = json.load(f)
+
+        for phase_name, external_id in self._gen_single_match_tactic_technique(mitre_json):
+            self.technique_to_tactics[external_id].append(phase_name)
+
+    def _gen_single_match_tactic_technique(self, mitre_json):
+        '''
+        Generator parsing the json from 'enterprise-attack.json',
+        and returning couples (phase_name, external_id)
+        '''
+        for obj in mitre_json.get('objects'):
+            if not obj.get('type') == 'attack-pattern':
+                continue
+            for e in obj.get('external_references'):
+                if not e.get('source_name') == 'mitre-attack':
+                    continue
+                external_id = e.get('external_id')
+                for kc in obj.get('kill_chain_phases'):
+                    if not kc.get('kill_chain_name') == 'mitre-attack':
+                        continue
+                    phase_name = kc.get('phase_name')
+                    yield (phase_name, external_id)
+
+    async def populate_data_directory(self, path_yaml=None):
+        '''
+        Populate the 'data' directory with the Atomic Red Team abilities.
+        These data will be usable by caldera after importation.
+        You can specify where the yaml files to import are located with the `path_yaml` parameter.
+        By default, read the yaml files in the atomics/ directory inside the Atomic Red Team repository.
+        '''
+        if not self.technique_to_tactics:
+            await self.populate_dict_techniques_tactics()
+
+        await self.init_payload_dir()
+
+        if not path_yaml:
+            path_yaml = os.path.join(self.repo_dir, 'atomics', '**', 'T*.yaml')
+
+        for filename in glob.iglob(path_yaml):
+            for entries in BaseWorld.strip_yml(filename):
+                for test in entries['atomic_tests']:
+                    try:
+                        incr_ingested, incr_total = await self._save_ability(entries, test)
+                        self.at_ingested += incr_ingested
+                        self.at_total += incr_total
+                    except Exception as e:
+                        self.log.debug('ERROR:', filename, e)
+                        self.errors += 1
+
+        # Reload data from the plugin, as we created abilities in the 'data' directory.
+        await self.data_svc.load_data(plugins=(self.plugin,))
```

Comment:
Mhhh it didn't work without the reload, I had to add it manually in 548c63d. The problem comes from [`c_plugin.enable()`](https://github.com/mitre/caldera/blob/master/app/objects/c_plugin.py#L49) (code below), called by [`AppService.load_plugins()`](https://github.com/mitre/caldera/blob/0b06fafc207b470b6b5cb06e1ba5e609fb6fddf1/app/service/app_svc.py#L86). When the `data` directory is created by the `enable()` method in plugin, it will not be taken into account.
```python
            if os.path.exists('plugins/%s/data' % self.name.lower()):
                self.data_dir = 'plugins/%s/data' % self.name.lower()
            plugin = self._load_module().enable
            await plugin(services)
```

Reply:
Not sure what the fix should be here. Should we double check the presence of a `data_dir`, before and after the plugin was loaded? Or should the plugin register its newly created `data_dir` to `data_svc`?

Reply:
I made a first fix in 3d83e3c

Reply:
I think we are over-complicating this a tad, how about the "keep-it-stupid-simple" approach - add an empty data directory with a `.gitkeep`?  That would resolve all of the issues you mentioned no?

Reply:
@zaphodef

Reply:
mhh that is true, gonna change this right away

Reply:
fixed in 32170dc :)